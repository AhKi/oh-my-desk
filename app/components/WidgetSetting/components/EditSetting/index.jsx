import React from 'react';
import PropTypes from 'prop-types';
import ToggleButton from 'components/Button/ToggleButton';

const propTypes = {
  item: PropTypes.shape({
    favicon: PropTypes.string,
    id: PropTypes.string,
    isActive: PropTypes.bool,
    isIcon: PropTypes.bool,
    isOnTop: PropTypes.bool,
    name: PropTypes.string,
    position: PropTypes.shape({
      x: PropTypes.number,
      y: PropTypes.number,
    }),
    size: PropTypes.shape({
      height: PropTypes.number,
      width: PropTypes.number,
    }),
    transparency: PropTypes.number,
    type: PropTypes.string,
    url: PropTypes.string,
  }),
  onChangeInput: PropTypes.func,
  onStoreWidgetInfo: PropTypes.func,
};
const defaultProps = {
  item: {},
  onChangeInput() {},
  onStoreWidgetInfo() {},
};

class EditSetting extends React.Component {
  constructor(props) {
    super(props);
    this.handleChangeName = this.handleChangeName.bind(this);
    this.handleChangeUrl = this.handleChangeUrl.bind(this);
    this.handleToggleIsActive = this.handleToggleIsActive.bind(this);
    this.handleToggleIsOnTop = this.handleToggleIsOnTop.bind(this);
  }

  handleChangeName(e) {
    this.props.onChangeInput('name', e.target.value);
  }

  handleChangeUrl(e) {
    this.props.onChangeInput('url', e.target.value);
  }

  handleToggleIsActive() {
    const { item, onStoreWidgetInfo } = this.props;
    onStoreWidgetInfo(
      item.id,
      Object.assign({}, item, { isActive: !item.isActive }),
    );
  }

  handleToggleIsOnTop() {
    const { item, onStoreWidgetInfo } = this.props;
    onStoreWidgetInfo(
      item.id,
      Object.assign({}, item, { isOnTop: !item.isOnTop }),
    );
  }

  render() {
    const { item } = this.props;

    return (
      <div className="EditSetting">
        <h4 className="EditSetting__header">Edit setting</h4>
        <div>
          <label htmlFor="name-input">
            Name
            <input
              id="name-input"
              type="text"
              value={item.name}
              onChange={this.handleChangeName}
            />
          </label>
          <label htmlFor="url-input">
            Url
            <input
              id="url-input"
              type="text"
              value={item.url}
              onChange={this.handleChangeUrl}
            />
          </label>
          <div>
            Activated
            <ToggleButton
              checkedValue={null}
              unCheckedValue={null}
              isCheck={item.isActive}
              onToggle={this.handleToggleIsActive}
            />
            Runs the Web widget.
          </div>
          <div>
            Always top
            <ToggleButton
              checkedValue={null}
              unCheckedValue={null}
              isCheck={item.isOnTop}
              onToggle={this.handleToggleIsOnTop}
            />
            Always pin the Web widget up.
          </div>
        </div>
      </div>
    );
  }
}

EditSetting.propTypes = propTypes;
EditSetting.defaultProps = defaultProps;

export default EditSetting;
