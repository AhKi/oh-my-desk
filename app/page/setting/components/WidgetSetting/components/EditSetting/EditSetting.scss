.EditSetting__content {
  margin-left: 30px;
}