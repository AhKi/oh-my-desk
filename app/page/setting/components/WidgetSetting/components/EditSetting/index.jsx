import React from 'react';
import PropTypes from 'prop-types';
import ToggleButton from 'setting/components/Button/ToggleButton';
import './EditSetting.scss';

const propTypes = {
  item: PropTypes.shape({
    favicon: PropTypes.string,
    id: PropTypes.string,
    isActive: PropTypes.bool,
    isIcon: PropTypes.bool,
    isOnTop: PropTypes.bool,
    name: PropTypes.string,
    position: PropTypes.shape({
      x: PropTypes.number,
      y: PropTypes.number,
    }),
    size: PropTypes.shape({
      height: PropTypes.number,
      width: PropTypes.number,
    }),
    transparency: PropTypes.number,
    type: PropTypes.string,
    url: PropTypes.string,
  }),
  onChangeInput: PropTypes.func,
  onStoreWidgetInfo: PropTypes.func,
};
const defaultProps = {
  item: {},
  onChangeInput() {},
  onStoreWidgetInfo() {},
};

class EditSetting extends React.Component {
  constructor(props) {
    super(props);
    this.handleChangeName = this.handleChangeName.bind(this);
    this.handleChangeUrl = this.handleChangeUrl.bind(this);
    this.handleToggleIsActive = this.handleToggleIsActive.bind(this);
    this.handleToggleIsOnTop = this.handleToggleIsOnTop.bind(this);
  }

  handleChangeName(e) {
    this.props.onChangeInput('name', e.target.value);
  }

  handleChangeUrl(e) {
    this.props.onChangeInput('url', e.target.value);
  }

  handleToggleIsActive() {
    const { item, onStoreWidgetInfo } = this.props;
    onStoreWidgetInfo(
      item.id,
      Object.assign({}, item, { isActive: !item.isActive }),
    );
  }

  handleToggleIsOnTop() {
    const { item, onStoreWidgetInfo } = this.props;
    onStoreWidgetInfo(
      item.id,
      Object.assign({}, item, { isOnTop: !item.isOnTop }),
    );
  }

  render() {
    const { item } = this.props;

    return (
      <div className="EditSetting space-6x">
        <p className="space-2x"><strong>Edit setting</strong></p>
        <div className="EditSetting__content">
          <label htmlFor="name-input" className="InputSet__label">
            <span className="InputSet__label-text">Name</span>
            <input
              className="InputSet InputSet__text-input"
              id="name-input"
              type="text"
              value={item.name}
              onChange={this.handleChangeName}
            />
          </label>
          <label htmlFor="url-input" className="InputSet__label">
            <span className="InputSet__label-text">Url</span>
            <input
              className="InputSet InputSet__text-input"
              id="url-input"
              type="text"
              value={item.url}
              onChange={this.handleChangeUrl}
            />
          </label>
          <div className="WidgetEdit__toggle-label">
            <span className="InputSet__label-text">Activated</span>
            <div className="WidgetEdit__toggle-btn">
              <ToggleButton
                checkedValue={null}
                unCheckedValue={null}
                isCheck={item.isActive}
                onToggle={this.handleToggleIsActive}
              />
              <span className="InputSet__toggle-description">Runs the Web widget.</span>
            </div>
          </div>
          <div className="WidgetEdit__toggle-label">
            <span className="InputSet__label-text">Always Top</span>
            <div className="WidgetEdit__toggle-btn">
              <ToggleButton
                checkedValue={null}
                unCheckedValue={null}
                isCheck={item.isOnTop}
                onToggle={this.handleToggleIsOnTop}
              />
              <span className="InputSet__toggle-description">Always pin the Web widget up.</span>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

EditSetting.propTypes = propTypes;
EditSetting.defaultProps = defaultProps;

export default EditSetting;
