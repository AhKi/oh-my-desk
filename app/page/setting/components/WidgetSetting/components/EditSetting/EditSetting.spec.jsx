import React from 'react';
import { shallow } from 'enzyme';

import EditSetting from '.';

describe('<EditSetting />', () => {
  it('should match to snapshot when render default', () => {
    const wrapper = shallow(
      <EditSetting
        item={{
          isOpen: false,
          isOnTop: false,
        }}
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should call onChangeInput when call handleChangeName', () => {
    const onChangeInput = jest.fn();
    const wrapper = shallow(
      <EditSetting
        item={{
          isOpen: false,
          isOnTop: false,
        }}
        onChangeInput={onChangeInput}
      />,
    );

    wrapper.instance().handleChangeName({
      target: {
        value: 'mock-value',
      },
    });

    expect(onChangeInput).toHaveBeenCalledTimes(1);
    expect(onChangeInput).toHaveBeenCalledWith('name', 'mock-value');
  });

  it('should call onChangeInput when call handleChangeUrl', () => {
    const onChangeInput = jest.fn();
    const wrapper = shallow(
      <EditSetting
        item={{
          isOpen: false,
          isOnTop: false,
        }}
        onChangeInput={onChangeInput}
      />,
    );

    wrapper.instance().handleChangeUrl({
      target: {
        value: 'mock-value',
      },
    });

    expect(onChangeInput).toHaveBeenCalledTimes(1);
    expect(onChangeInput).toHaveBeenCalledWith('url', 'mock-value');
  });

  it.skip('should call onStoreWidgetInfo when call handleToggleIsActive', () => {
    const onStoreWidgetInfo = jest.fn();
    const wrapper = shallow(
      <EditSetting
        item={{
          id: 'mock-id',
          isOpen: false,
          isOnTop: false,
        }}
        onStoreWidgetInfo={onStoreWidgetInfo}
      />,
    );

    wrapper.instance().handleToggleIsActive();

    expect(onStoreWidgetInfo).toHaveBeenCalledTimes(1);
    expect(onStoreWidgetInfo).toHaveBeenCalledWith(
      'mock-id',
      {
        id: 'mock-id',
        isOpen: true,
        isOnTop: false,
      },
    );
  });

  it.skip('should call onStoreWidgetInfo when call handleToggleIsOnTop', () => {
    const onStoreWidgetInfo = jest.fn();
    const wrapper = shallow(
      <EditSetting
        item={{
          id: 'mock-id',
          isOpen: false,
          isOnTop: false,
        }}
        onStoreWidgetInfo={onStoreWidgetInfo}
      />,
    );

    wrapper.instance().handleToggleIsOnTop();

    expect(onStoreWidgetInfo).toHaveBeenCalledTimes(1);
    expect(onStoreWidgetInfo).toHaveBeenCalledWith(
      'mock-id',
      {
        id: 'mock-id',
        isOpen: false,
        isOnTop: true,
      },
    );
  });
});
