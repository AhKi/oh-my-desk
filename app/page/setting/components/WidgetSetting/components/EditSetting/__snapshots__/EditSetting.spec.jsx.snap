// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditSetting /> should match to snapshot when render default 1`] = `
<div
  className="EditSetting"
>
  <h4
    className="EditSetting__header"
  >
    Edit setting
  </h4>
  <div>
    <label
      htmlFor="name-input"
    >
      Name
      <input
        id="name-input"
        onChange={[Function]}
        type="text"
      />
    </label>
    <label
      htmlFor="url-input"
    >
      Url
      <input
        id="url-input"
        onChange={[Function]}
        type="text"
      />
    </label>
    <div>
      Activated
      <ToggleButton
        checkedValue={null}
        isCheck={false}
        onToggle={[Function]}
        unCheckedValue={null}
        width={50}
      />
      Runs the Web widget.
    </div>
    <div>
      Always top
      <ToggleButton
        checkedValue={null}
        isCheck={false}
        onToggle={[Function]}
        unCheckedValue={null}
        width={50}
      />
      Always pin the Web widget up.
    </div>
  </div>
</div>
`;
